functions.py

```diff
@@ -50,20 +47,14 @@ def sama_counter(week):
             gromlinvip_count+=1
     return moonsama_count, exosama_count, gromlinvip_count
 
-def sama_lister(week):
-    from inputs import raw_gganbu_array, raw_result_array
+def attendance_counter(first_week, last_week):
+    from inputs import raw_result_array
     week_index = week - 17
-    gganbu_dictionary = get_and_load_dictionary(raw_gganbu_array[week_index])
-    result_dictionary = get_and_load_dictionary(raw_result_array[week_index])
-    player_list = []
-    for player in result_dictionary.keys():
-        player_list.append(player)
-    return player_list
+    rss_per_user_week_dict = get_and_load_dictionary(raw_result_array[week_index])
 
-def attendance_counter(first_week, last_week):
     attendance_log = []
     for week in range(first_week, last_week+1):
-        for player in sama_lister(week):
```

Comment:
Look here. samalister() fetches all all player data and gganbu data for a given week. this means that for every player, we are refetching all the player data and gganbu data for a given week, which is redundant.